### DISCUSSION TITLE
Real-Debrid Client Updates

### DISCUSSION THREAD

**Discussion by GameEnder** [tteck/Proxmox#2761] [Ideas]
Can the update process be looked in to for Real-Debrid Client. It only updates from a file replacement from the latest release and the induced powershell update script dose not work out of the box. You need powershell installed in the container and the script need to be fixed to work on Linux Powershell.

Fixed Script

```
$currentDirectory = $PSScriptRoot

Write-Host "Starting update script in $currentDirectory"

if ($PSVersionTable.Platform -eq "Unix") {
    if ((id -u) -eq 0) {


Write-Host "Stopping ReadDebridClient..."

systemctl stop rdtc.service

Write-Host "Stopped ReadDebridClient"

$releasesUri = "https://api.github.com/repos/rogerfar/rdt-client/releases/latest"
$downloadUri = ((Invoke-RestMethod -Method GET -Uri $releasesUri).assets | Where-Object name -like "*.zip").browser_download_url

Write-Host "Downloading $downloadUri"

$pathZip = Join-Path -Path $([System.IO.Path]::GetTempPath()) -ChildPath $(Split-Path -Path $downloadUri -Leaf)

Invoke-WebRequest -Uri $downloadUri -Out $pathZip

$tempExtract = Join-Path -Path $([System.IO.Path]::GetTempPath()) -ChildPath $((New-Guid).Guid)

Write-Host "Extracting to $tempExtract"

Expand-Archive -Path $pathZip -DestinationPath $tempExtract -Force

Write-Host "Backing up appsettings.json"

Copy-Item -Path "$currentDirectory\appsettings.json" -Destination $tempExtract -Force 

Write-Host "Moving new files"

Move-Item -Path "$tempExtract\*" -Destination $currentDirectory -Force

Write-Host "Removing temp files"

Remove-Item -Path $tempExtract -Force -Recurse -ErrorAction SilentlyContinue

Remove-Item $pathZip -Force

Write-Host "Starting ReadDebridClient..."

systemctl start rdtc.service

Write-Host "Started ReadDebridClient"

    }

    else {
        exit
    }
}
```

**Comment by tteck** [comment]
So, the current update method does not work?

**Reply by GameEnder** [reply]
It updates the OS just fine just not Real-Debrid Client. The Real-Debrid Client documentation says to use the Update powershell script if you are using a non docker install. But the script was setup for Windows not Linux.

**Reply by tteck** [reply]
The update process has nothing to do with the OS

https://github.com/tteck/Proxmox/blob/2847d9cc9a121c2eba55cccc93b1413f1a3a652a/ct/rdtclient.sh#L55-L69

**Reply by GameEnder** [reply]
Sorry I was meaning your LXC updater script that dose OS updates. This is the only application from your scripts that I use that dose not update from the Distribution repos or from it own builtin updater. Dose the updater script update other applications that are not in the distributions repos?

**Reply by tteck** [reply]
Type the command `update` in the LXC console.

**Reply by GameEnder** [reply]
Oh really... I am dumb. I was not even aware that was a feature. Thanks for the help.
